code/game/machinery/computer/robot.dm

```diff
@@ -132,3 +181,13 @@
 					s.start()
 					D.visible_message("<span class='danger'>\the [D] self-destructs!</span>")
 					D.gib()
+		if("extract")
+			if(!GLOB.upload_code)
+				GLOB.upload_code = random_nukecode()
+
+			message_admins("[ADMIN_LOOKUPFLW(usr)] is extracting the upload key!")
+			extracting = TRUE
+			ui_update()
+			extraction(usr)
```

Comment:
Do this asynchronously or use timers instead

Reply:
Asynchronously?, Im probably just going to use a timer, ill look at this tomorrow